### DISCUSSION TITLE
Is it possible to have pathnames work without a prefix, even when it is not the default locale?

### DISCUSSION THREAD

**Discussion by BearCooder** [amannn/next-intl#531] [Q&A] [answered]
Hi,

Lets say I have this piece in my middleware:

```
export default createMiddleware({
  // A list of all locales that are supported
  locales: ['de', 'en'],

  // If this locale is matched, pathnames work without a prefix (e.g. `/about`)
  defaultLocale: 'de',
});
```

For now when the locale is de or someone switches the language to de the url works without the de prefix. So for example I have only
http://localhost:3000/

But when the locale is en then it will be http://localhost:3000/en

Is there a way to have it work without the prefix even for not the default locale? To have a "clean" url? (like in this example on [github](https://github.com/anthonyshew/i18n-demo))

**Comment by amannn** [comment, marked as answer]
Are you looking for this option? https://next-intl-docs.vercel.app/docs/routing/middleware#never-use-a-locale-prefix

**Reply by BearCooder** [reply]
Very nice I did not see this in the documentation! By reading it, it looks very straightforward to do it. Thank you very much!
I will later try it out and if its working out of the box then I mark it as answer :)
